packages/crypto/src/lib/crypto.ts

```diff
@@ -372,6 +373,103 @@ async function getAmdCert(url: string): Promise<Uint8Array> {
   }
 }
 
+export const walletEncrypt = async(
+  myWalletSecretKey: Uint8Array, 
+  theirWalletPublicKey: Uint8Array,
+  sessionSig: Uint8Array,
+  message: Uint8Array
+): Promise<WalletEncryptedPayload> => {
+  const random = new Uint8Array(16);
+  window.crypto.getRandomValues(random);
+  const dateNow = Date.now();
+  const createdAt = Math.floor(dateNow / 1000);
+  const timestamp = Buffer.alloc(8);
+  timestamp.writeBigUInt64BE(BigInt(createdAt), 0);
+
+  const myWalletPublicKey = new Uint8Array(32);
+  nacl.crypto_scalarmult_base(myWalletPublicKey, myWalletSecretKey);
```

Comment:
nit: if we can use @noble curves and hashes way better so we move towards lib unification (and drop tweetnacl)